crates/bevy_color/src/hsla.rs

```diff
@@ -150,6 +146,24 @@ impl From<bevy_render::color::Color> for Hsla {
     }
 }
 
+impl From<LinearRgba> for Hsla {
+    fn from(value: LinearRgba) -> Self {
+        Srgba::from(value).into()
+    }
+}
+
+impl From<Oklaba> for Hsla {
+    fn from(value: Oklaba) -> Self {
+        Srgba::from(value).into()
```

Comment:
We need to be careful here, because some conversions internally convert from linear to srgb - so you'd be converting linear to srgb, and then back to linear, before converting to the target space.

I think we should either guarantee that the conversion path between two spaces is the optimal one, or we should remove support for direct conversions between those two spaces - a signal to the user that doing the conversion involves extra work.

If you look closely at the conversion for LCH, it's actually doing multiple "hops": srgba -> linear -> XYZ -> Lab -> LCH. Ideally, the `From` implementation should make this explicit, that is, it's leveraging the `From` for those other color spaces rather than duplicating the code.

Reply:
I do agree that these conversions should take the absolute shortest path, but I think we should provide them even if they are via an intermediary space. Once this lands (if it does) I will then add XYZ and Lab, which should allow for all the conversion algorithms to be small and explicit.